File: core/src/com/inkostilation/pong/notifications/INotifier.java
package com.inkostilation.pong.notifications;

import com.inkostilation.pong.engine.Ball;
import com.inkostilation.pong.engine.Field;
import com.inkostilation.pong.engine.Paddle;

public interface INotifier {

    <O> void notifyObservers(O... observable);
    <O> void subscribe(IObserver<O> observer, Class<O> type);
    <O> void unsubscribe(IObserver<O> observer, Class<O> type);
}

File: core/src/com/inkostilation/pong/notifications/INotifier.kt
package com.inkostilation.pong.notifications

interface INotifier {
    fun <O> notifyObservers(vararg observable: O)
    fun <O> subscribe(observer: IObserver<O>?, type: Class<O>?)
    fun <O> unsubscribe(observer: IObserver<O>?, type: Class<O>?)
}